security/pkg/k8s/tokenreview/k8sauthn.go

```diff
@@ -16,22 +16,56 @@ package tokenreview
 
 import (
 	"context"
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
 	"strings"
 
 	k8sauth "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 
-	"istio.io/istio/pkg/jwt"
+	"istio.io/pkg/env"
+	"istio.io/pkg/log"
 )
 
-const (
+var (
 	// The default audience for SDS trustworthy JWT. This is to make sure that the CSR requests
 	// contain the JWTs intended for Citadel.
-	DefaultAudience = "istio-ca"
+	DefaultAudience = env.RegisterStringVar("TOKEN_AUDIENCE", "istio-ca", "Audience to check in accepted JWTW tokens")
+	RequireAudience = env.RegisterBoolVar("REQUIRE_AUDIENCE", false, "Reject tokens without audience. If false, K8S token will be accepted")
```

Comment:
Normally audience should be required. Is that for 1st party JWT? If so, the default behavior should be "require audience". Audience check is skipped only for special cases.

Reply:
The audience is always checked for 3p tokens. 

This PR allows users with 1p tokens - so there is an upgrade path. Some clusters don't support or don't enable 3p tokens.

In practice any 1p token can be trivially converted to a 3p token - so from a security point of view if someone gets hold of
a 1p token (no audience) they can convert it to a 3p token. 

I'll rename the flag to be more clear - ALLOW_K8S_DEFAULT_TOKEN maybe ?